### DISCUSSION TITLE
Home assistant : Perchè?

### DISCUSSION THREAD

**Discussion by Giovanni5860** [indomus/forum#153] [HUB: Home Assistant]
Saluti a tutto il Forum.
Scusate la domanda provocatoria; mi accingo a dotare casa di alcune funzioni domotiche : Automatizzare Luci,tapparelle, impianto riscaldamento, condizionatori e Tv con broadlink, il tutto integrato ad Alexa. La mia scelta di base è orientata su dispositivi Shelly. Ho visto che il tutto è integrabile utilizzando i vari software proprietari con un discreto livello di integrazione e automazione. A questo punto la domanda : **che vantaggi aggiuntivi otterrei utilizzando Home assistant?**
Da 'anziano' programmatore sono fortemente tentato di acquistare un raspberry e cimentarmi nel progetto, ma rimane sempre il dubbio di fondo; ne vale la pena? E in termini di manuntenzione : se si dovesse bloccare l'applicazione e io, per un motivo qualsiasi, non potessi intervenire, chi sarebbe in grado di mettere mani? Al contario delle varie App proprietarie la cui manutenzione è più intuitiva. Grazie per l'attenzione,attendo riflessioni e suggerimenti sull'argomento.

**Comment by snakuzzo** [comment]
Ciao...se parliamo di integrazione...confrontando le app proprietarie ed Home Assistant (o comunque uno qualsiasi dei più importanti hub personali), non c'è proprio storia.
Con un hub personale hai la possibilità di integrare le più disparate tipologie di device, creando scene, script, automazioni, sensori custom basati su dati  che coinvolgano tutti i tuoi device...indipendentemente dal costruttore....insomma...puoi creare qualsiasi cosa tu voglia. Pensa ad una cosa e puoi farla. Ovvio che in base alla complessità poi dovrai studiare e capire come fare, ma hai una libertà totale.
Per utilizzare HA non serve saper programmare. Di base non ti serve proprio niente...ti basterebbe leggere e capire come funziona il "prodotto". Se poi vuoi fare qualcosa di più complesso ti bastano poche cavolate, da conoscere più a livello di logica che di sintassi (es. i cicli) e, visto che dici di essere un "anziano programmatore", non dovresti incontrare difficoltà. Per la manutenzione, una volta capito come funziona, a mio avviso ti basta sapere leggere il file di log (è la PRIMISSIMA cosa da fare...sempre).
Considera che Home Assistant non è un prodotto "chiavi in mano", non è una cosa che installi, configuri e sta lì per sempre.
Va seguito, non dico giorno dopo giorno, ma devi stargli dietro considerando che ormai viene pubblicata una release al mese che può implicare o meno i cosiddetti "breaking change" legati alle tue integrazioni. In tal caso dovrai leggere e sistemare la configurazione.
Insomma...soprattutto all'inizio devi dedicargli un bel po' di tempo.

Nessuno deve intervenire al posto tuo...sei tu il padrone, ma se non sposi questo modo di pensare e dici che non hai tempo per seguirlo o per metterti a studiare, il consiglio è quello di rimanere con la domotica fatta con Google e Alexa.

**Reply by Giovanni5860** [reply]
Grazie per la bella risposta (e anche per il tempo che mi hai dedicato).
Ho letto un pò di documentazione su Ha e mi sono ben chiare le potenzialità. Non ho dubbi sul fatto che se dedichi del tempo a un progetto come questo ottieni risultati entusiasmanti. Il fatto è che probabilmente l'impegno è esagerato per quello che intendo ottenere in prima battuta (luci,tapparelle,tv). Il fatto che poi 'IO'debba essere l'unico proprietario mi lascia ancora più perplesso (quando dico 'anziano'  vuol dire anche non sono più un giovanotto e l'idea di lasciare ai posteri un sistema 'chiuso' non mi entusiasma).Da un punto di vista pratico l'unico vantaggio che vedo è che HA funziona anche in assenza di connessione internet e che,sicuramente, è meno esposto a intrusioni indesiderate. In ogni caso potrei partire con una soluzione semlice per poi,all'occorrenza, implementare un sistema basato su HA.
Saluti,giovanni
